CustomerRegister/Controllers/CustomerController.cs

```diff
@@ -0,0 +1,91 @@
+﻿using CustomerRegister.Repositories.Interfaces;
+using Microsoft.AspNetCore.Mvc;
+using System;
+
+namespace CustomerRegister.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class CustomerController : ControllerBase
+    {
+        private readonly ICustomerService _customerService;
+
+        public CustomerController(ICustomerService customer)
+        {
+            _customerService = customer ?? throw new ArgumentNullException(nameof(customer));
+        }
+
+        [HttpPost]
+        public IActionResult Create([FromBody] CustomerEntity customer)
+        {
+            try
+            {
+                var customerId = _customerService.Add(customer);
+                return Created("Id: ", customerId);
+            }
+            catch (ArgumentException e)
+            {
+                return BadRequest(e.Message);
+            }
+        }
+
+        [HttpGet]
+        public IActionResult GetAllCustomers()
+        {
+            try
+            {
+                var response = _customerService.GetAllCustomers();
+                return Ok(response);
+            }
+            catch
+            {
+                return NoContent();
+            }
```

Comment:
Apenas devemos retornar 204 quando o endpoint não possui retorno o que nunca vai ser o caso aqui, como estamos fazendo GetAll ou vai retornar os dados ou uma lista vazia quando não tiver nenhum dado, ou seja, possui retorno.
```suggestion
            try
            {
                var listCustomers = _service.GetAll();
                return Ok(listCustomers);
            }
            catch (Exception ex)
            {
                var exMessage = ex.InnerException?.Message ?? ex.Message;
                return Problem(exMessage);
            }
```

Reply:
Feito